searvey/usgs.py

```diff
@@ -400,8 +364,9 @@ def get_usgs_data(
             # in `data_retrieval` due to non-existant data
             continue
 
-        ds = _get_dataset_from_query_results(result.result, usgs_metadata, truncate_seconds)
-        datasets.append(ds)
+        ds = _get_dataset_from_query_results(result.result, usgs_metadata)
+        if len(ds) > 0:
```

Comment:
I think that `len(ds)` returns the number of data variables. You can have a dataset with data variables and no data (e.g. if you filter them out using `.sel()`). Obviously this is related with the changes you did at lines 294-295. Nevertheless I would investigate if it is possible to use the dimensions to identify an empty dataset. E.g. `ds.dims["site_no"] == 0`

Reply:
Depending on the answer to https://github.com/oceanmodeling/searvey/pull/90#discussion_r1232202089 I can either check for dataset variable length as you suggested or this should just work fine.